utils_linux.go

```diff
@@ -247,11 +251,19 @@ func (r *runner) run(config *specs.Process) (_ int, retErr error) {
 		}
 		process.ExtraFiles = append(process.ExtraFiles, os.NewFile(uintptr(i), "PreserveFD:"+strconv.Itoa(i)))
 	}
-	detach := r.detach || (r.action == CT_ACT_CREATE)
 	// Setting up IO is a two stage process. We need to modify process to deal
 	// with detaching containers, and then we get a tty after the container has
 	// started.
-	handlerCh := newSignalHandler(r.enableSubreaper)
+	if r.enableSubreaper {
+		// set us as the subreaper before registering the signal handler for the container
+		if err := system.SetSubreaper(1); err != nil {
```

Comment:
Opening a new comment, but it is the same that in the resolved one, just that it got mixed with other things.

Do we want to set the subreaper even on detached containers?

Reply:
In fact I don't know, but as I said before, I want to keep it the same as before.

Reply:
Oh, right, this is the same as before. Okay, thanks!